src/png/mod.rs

```diff
@@ -383,6 +390,10 @@ impl PngImage {
                                 best_size = size;
                                 std::mem::swap(&mut best_line, &mut f_buf);
                                 best_line_raw.clone_from(&line_data);
+                                if size == 0 {
+                                    // Best possible result
+                                    break;
+                                }
```

Comment:
For any filter other than `Entropy` or `BigEnt`, could we initialize the `best_possible` variable to match the corresponding right-hand value in these comparisons? I think that approach would be slightly cleaner, as it would make every "best possible result" check the same and centralize the definition of the best possible entropy in one place.

Reply:
We could do that, it would just require more type casts - Entropy and BigEnt use i32 while MinSum and Bigrams use usize.

Reply:
Given that `as _` can be used to concisely let the Rust compiler figure out the target data type here, I think I still prefer my suggestion.

Reply:
That also works! :+1: